stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewGroupDescriptor.java

```diff
@@ -27,8 +28,8 @@
    * we emit the Fragment instead, and then let the Fragment's descriptor emit the View as its sole
    * child. This allows us to see Fragments in the inspector as part of the UI tree.
    */
-  private final Map<View, Object> mViewToElementMap =
-      Collections.synchronizedMap(new WeakHashMap<View, Object>());
+  private final Map<View, WeakReference<Object>> mViewToElementMap =
+      Collections.synchronizedMap(new WeakHashMap<View, WeakReference<Object>>());
```

Comment:
Using a `WeakReference` for the value will allocate an extra object for every single `View` that's present in the view hierarchy. Instead of always using a WR for the value, why not use a sentinel object to indicate "the value is the same as the key" (which is always the case when the value is a `View`). And then only use a WR for fragments (of which there are very few). This will save the allocations from always using a WR, while also avoiding the memory leak you discovered.

```
private final Object mViewSentinelValue = new Object();
...
Object elementReference = mViewToElementMap.get(childView);
if (elementReference == mViewSentinelValue) {
  return childView;
} else if (elementReference instanceof WeakReference) { 
  ...
}
...
mViewToElementMap.put(childView, mViewSentinalValue);
```